server/models/Earnings.js

```diff
@@ -200,18 +200,24 @@ const getEarnings =
 
       if (filterCriteria?.grower || filterCriteria?.phone) {
         const filterFormattedEarnings = formattedEarnings
-          .filter(({ grower }) =>
-            filterCriteria?.grower
+          .filter(({ grower }) => {
+            if (!grower && filterCriteria?.grower) {
+              return false;
+            }
```

Comment:
we must check the case that there isn't the criteria grower, if not, it would throw error  "read undefined's property: toLowerCase()"